src/blockchain/parser/index.rs

```diff
@@ -130,15 +130,15 @@ impl fmt::Debug for BlockIndexRecord {
 pub fn get_block_index(path: &Path) -> OpResult<HashMap<u64, BlockIndexRecord>> {
     info!(target: "index", "Reading index from {} ...", path.display());
 
-    let mut block_index = HashMap::with_capacity(800000);
+    let mut block_index = HashMap::with_capacity(900000);
     let mut db_iter = DB::open(path, Options::default())?.new_iter()?;
     let (mut key, mut value) = (vec![], vec![]);
 
     while db_iter.advance() {
         db_iter.current(&mut key, &mut value);
         if is_block_index_record(&key) {
             let record = BlockIndexRecord::from(&key[1..], &value)?;
-            if record.status & (BLOCK_VALID_CHAIN | BLOCK_HAVE_DATA | BLOCK_VALID_CHAIN) > 0 {
+            if record.status & (BLOCK_VALID_CHAIN | BLOCK_HAVE_DATA) > 0 {
```

Comment:
Is `BLOCK_VALID_CHAIN` not needed?

Reply:
`BLOCK_VALID_CHAIN` is already the first operand to the or expression. `x || x == x`.

Reply:
Ah yes, ofc. Missed that it was there two times